streams/shared/src/main/scala/zio/stream/experimental/ZChannel.scala

```diff
@@ -621,6 +621,59 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
     self >>> reader
   }
 
+  def mapOutMPar[Env1 <: Env, OutErr1 >: OutErr, OutElem2](n: Int)(
+    f: OutElem => ZIO[Env1, OutErr1, OutElem2]
+  ): ZChannel[Env1, InErr, InElem, InDone, OutErr1, OutElem2, OutDone] =
+    ZChannel.managed {
+      ZManaged.withChildren { getChildren =>
+        for {
+          _           <- ZManaged.finalizer(getChildren.flatMap(Fiber.interruptAll(_)))
+          queue       <- Queue.bounded[ZIO[Env1, Either[OutErr1, OutDone], OutElem2]](n).toManaged(_.shutdown)
+          errorSignal <- Promise.makeManaged[OutErr1, Nothing]
+          permits     <- Semaphore.make(n.toLong).toManaged_
+          pull        <- self.toPull
+          _ <- pull
+                 .foldM(
```

Comment:
A `Die` will make this deadlock. So we should use `foldCauseM` here.

Reply:
Good point, thanks.